### ISSUE TITLE
Infinity open dialog on exiting

### ISSUE THREAD

**Issue by epheien** [macvim-dev/macvim#869]
Reproduce:
```
gvim -u NONE -U NONE
:setlocal bufhidden=wipe colorcolumn= nobuflisted nocursorcolumn nocursorline nolist nonumber norelativenumber nospell noswapfile matchpairs=
ihello<Esc>
:setlocal nomodifiable nomodified
:terminal
```
and then, click window's 'close' button with mouse, will has a dialog
```
Question
Kill job in "!/bin/bash"?
```
click 'yes'

now, alway open a file open dialog, expect select a file.
and has not any information about this dialog

**Comment by ychin** [comment]
Reproduced the result. Will look into it. Thanks for filing the bug.

**Comment by ichizok** [comment]
Simplified reproducible steps:

```
:setlocal nobuflisted
:terminal
```

**Comment by ichizok** [comment]
Fixed by upstream [v8.1.1594](https://github.com/vim/vim/commit/2eed9bafa013f278ffe8bcd52489f90b11cf00e1)

**Comment by eirnym** [comment]
This issue is not reproducible anymore in snapshot 157.
